GeneralsMD/Code/GameEngine/Source/Common/RTS/TunnelTracker.cpp

```diff
@@ -301,6 +322,34 @@ void TunnelTracker::healObject( Object *obj, void *frames)
 	}
 }
 
+void TunnelTracker::updateFullHealTime()
+{
+	UnsignedInt minFrames = ~0u;
+
+	for (std::list<ObjectID>::const_iterator it = m_tunnelIDs.begin(); it != m_tunnelIDs.end(); ++it)
+	{
+		const Object* tunnelObj = TheGameLogic->findObjectByID(*it);
+		if (!tunnelObj)
+			continue;
+
+		const ContainModuleInterface* contain = tunnelObj->getContain();
+		DEBUG_ASSERTCRASH(contain != NULL, ("Contain module is NULL"));
+
+		if (!contain || !contain->isTunnelContain())
```

Comment:
`!contain` test is superfluous after assert.

Reply:
The assert won't stop the game crashing though.